Civi/Core/ClassScanner.php

```diff
@@ -140,8 +140,7 @@ private static function scanCoreClasses(): array {
     // TODO: Consider expanding this search.
     $classes = [];
     static::scanFolders($classes, $civicrmRoot, 'Civi/Test/ExampleData', '\\');
-    static::scanFolders($classes, $civicrmRoot, 'CRM/*/WorkflowMessage', '_');
-    static::scanFolders($classes, $civicrmRoot, 'CRM/*/Import', '_');
+    static::scanFolders($classes, $civicrmRoot, 'CRM', '_', ';(Upgrade|Utils);');
```

Comment:
@totten I'm honestly not sure if I got this regex right. I'm not sure how to test it to see if it really does exclude those directories.

Reply:
@colemanw OK, so here's a way to compare the list of scanned folders before/after some patch:

https://gist.github.com/totten/6ba1b077d404bf12850023d911216c2f

I would also suggest skipping the QF stuff and some other low-level bits, ie

```php
   static::scanFolders($classes, $civicrmRoot, 'CRM', '_', ';(Upgrade|Utils|Exception|_DAO|_Page|_Form|_Controller|_StateMachine|_Selector|_CodeGen);');
   static::scanFolders($classes, $civicrmRoot, 'Civi', '\\', ';\\\(Security|Test|Page|Form)\\\;');
```

These are classes where the *meaning* of service-registration would be... strange. (It feels awkward to fire listeners on a `Page` class when the `Page` class isn't in use.) And it limits the impact wrt total# scans.

| Exclusion Rule | Total Scanned Classes |
| -- | -- |
| Current `master` | 643 |
| PR (as currently written) | 1950 (+200%) |
| PR (with proposed exclusions) | 1034 (+60%) |

(*Aside: It's kind of hard for me to reason about benchmarking -- in theory, the cost of a large class-scan would be apparent on a shared server with cold opcode cache; but not so apparent on a fast desktop with no concurrent work and warm class-cache. It would be laborious to evaluate the difference between those. But 200% and 60% seem like big differences...*)